app/controllers/users_controller.rb

```diff
@@ -67,8 +67,11 @@ def handle_ip
     # Prevent someone from gaming the site by referring themselves.
     # Presumably, users are doing this from the same device so block
     # their ip after their ip appears three times in the database.
+    # Using split.first in case client is behind proxy to get his ip.
+    # If app is deployed on standalone server use request.remote_ip to
+    # get client ip address.
 
-    address = request.env['HTTP_X_FORWARDED_FOR']
+    address = request.env['HTTP_X_FORWARDED_FOR'].try(:split).try(:first) || request.remote_ip
```

Comment:
You are absolutely right about X-Forwarded-For comma-space separation, that was a mistake.
What about request.remote_ip, it probably should walk around proxies, but on Heroku it returns Heroku router ip address instead of client's. http://stackoverflow.com/questions/18264304/get-clients-real-ip-address-on-heroku
Despite there's mentioned that client's remote ip address will be the last one, actually I've got Heroku + Cloudflare CDN returning client's remote ip the first one in sequence.